pkg/storage/chunk/cache/resultscache/cache_test.go

```diff
@@ -682,6 +682,107 @@ func Test_resultsCache_MissingData(t *testing.T) {
 	require.False(t, hit)
 }
 
+func Test_shouldCacheReq(t *testing.T) {
+	modelNow := model.Now()
+
+	cfg := Config{
+		CacheConfig: cache.Config{
+			Cache: cache.NewMockCache(),
+		},
+	}
+	c, err := cache.New(cfg.CacheConfig, nil, log.NewNopLogger(), stats.ResultCache, constants.Loki)
+	require.NoError(t, err)
+	rc := NewResultsCache(
+		log.NewNopLogger(),
+		c,
+		nil,
+		ConstSplitter(day),
+		mockLimits{},
+		MockMerger{},
+		MockExtractor{},
+		nil,
+		nil,
+		func(_ context.Context, tenantIDs []string, r Request) int {
+			return 10
+		},
+		nil,
+		false,
+		false,
+	)
+	require.NoError(t, err)
+
+	// create cache with handler
+	ctx := user.InjectOrgID(context.Background(), "1")
+
+	// create request with start end within the key extents
+	req := parsedRequest.WithStartEndForCache(time.UnixMilli(int64(modelNow)-(50*1e3)), time.UnixMilli(int64(modelNow)-(10*1e3)))
+
+	// fill cache
+	key := ConstSplitter(day).GenerateCacheKey(context.Background(), "1", req)
+	rc.put(ctx, key, []Extent{mkExtent(int64(modelNow)-(600*1e3), int64(modelNow))})
```

Comment:
this could be flaky at times, because the extent might end up in the previous day when this test runs on a day boundary

Reply:
we could use constant time range similar to other tests to keep this predictable? `mkExtent(50, 120)`

Reply:
extent should be irrelevent here I think because, the test is make sure we are calling upstream `next` handler instead of looking up in the cache in the first place (independent of cache hit or cache miss).

But good idea to make it constant anyway. Will update it.